resources/functionMetadata.php

```diff
@@ -837,13 +844,25 @@
 	'finfo::buffer' => ['hasSideEffects' => false],
 	'finfo::file' => ['hasSideEffects' => false],
 	'floatval' => ['hasSideEffects' => false],
+	'flock' => ['hasSideEffects' => true],
 	'floor' => ['hasSideEffects' => false],
 	'fmod' => ['hasSideEffects' => false],
+	'fopen' => ['hasSideEffects' => true],
+	'fpassthru' => ['hasSideEffects' => true],
+	'fputcsv' => ['hasSideEffects' => true],
+	'fputs' => ['hasSideEffects' => true],
+	'fread' => ['hasSideEffects' => true],
+	'fscanf' => ['hasSideEffects' => true],
+	'fseek' => ['hasSideEffects' => true],
+	'fstat' => ['hasSideEffects' => true],
+	'ftell' => ['hasSideEffects' => true],
 	'ftok' => ['hasSideEffects' => false],
+	'ftruncate' => ['hasSideEffects' => true],
 	'func_get_arg' => ['hasSideEffects' => false],
 	'func_get_args' => ['hasSideEffects' => false],
 	'func_num_args' => ['hasSideEffects' => false],
 	'function_exists' => ['hasSideEffects' => false],
+	'fwrite' => ['hasSideEffects' => true],
```

Comment:
this reminds me of, whether `file_put_contents` and `file_get_contents` is missing in this list?

Reply:
I only included the functions that work on file handles. But I guess all filesystem operations probably should be added. In that case it should also include `unlink()`, `file()`, `mkdir()`, and tons of others. I'll try to do that and see if anything breaks.